### PR TITLE
MS 42: Admin refactor

### PR DESCRIPTION
**Pull request by amacazzaga**

### CODE REVIEW

File: controllers/admins.js [hunk 1]
```diff
@@ -2,48 +2,37 @@ const fs = require('fs');
 const admins = JSON.parse(fs.readFileSync('./data/admins.json'));
 
 const getAll = (req, res) => {
-  res.json(admins);
-};
+  return res.status(200).json(admins)
```
Comment (nicolasblasco) [inline controllers/admins.js:5]: @amacazzaga you forgot to add ;

File: controllers/admins.js [hunk 2]
```diff
@@ -2,48 +2,37 @@ const fs = require('fs');
...
+      admin.username = updateAdmin.username ? updateAdmin.username : admin.username;
+      admin.password = updateAdmin.password ? updateAdmin.password : admin.password;
+      return res.status(200).json(admin);
+    }
+  });
+  } 	return res.status(404).send("Admin not found");
+  };
 
 module.exports = {
   getAll: getAll,
```
Comment (nicolasblasco) [inline controllers/admins.js:38]: @amacazzaga you can simplify module exports, for example:
module.exports = {
    getAll,
    getById,
    getbyName,
    update
}

File: controllers/admins.js [hunk 3]
```diff
@@ -2,52 +2,41 @@ const fs = require('fs');
 const admins = JSON.parse(fs.readFileSync('./data/admins.json'));
 
 const getAll = (req, res) => {
-  res.json(admins);
-};
+  return res.status(200).json(admins);
+  };
```
Comment (NicoMunoz909) [inline controllers/admins.js:6]: Remove indentation here

File: controllers/admins.js [hunk 4]
```diff
@@ -2,52 +2,41 @@ const fs = require('fs');
...
 const getById = (req, res) => {
   const admin = admins.find(admin => admin.id === req.params.id);
-  
-  if (admin) {
-    res.json(admin);
-  } else {
-    res.status(400).send('Admin not found');
+  if (admin) return res.status(200).json(admin);
+  return res.status(404).send("Admin not found");
   }
```
Comment (NicoMunoz909) [inline controllers/admins.js:12]: Also remove indentation here

File: controllers/admins.js [hunk 5]
```diff
@@ -2,52 +2,41 @@ const fs = require('fs');
...
 const getByName = (req, res) => {
   const admin = admins.find(admin => admin.name === req.params.name);
-
-  if (admin) {
-    res.json(admin);
-  } else {
-    res.status(400).send('Admin not found');
+  if (admin) return res.status(200).json(admin);
+  return res.status(404).send("Admin not found");
   }
```
Comment (NicoMunoz909) [inline controllers/admins.js:18]: Also here

File: controllers/admins.js [hunk 6]
```diff
@@ -2,52 +2,41 @@ const fs = require('fs');
...
-  }
-};
+  const updateAdmin = req.query;
+  admins.forEach(admin => {
+    if (admin.id === req.params.id) {
+      admin.name = updateAdmin.name ? updateAdmin.name : admin.name;
+      admin.email = updateAdmin.email ? updateAdmin.email : admin.email;
+      admin.username = updateAdmin.username ? updateAdmin.username : admin.username;
+      admin.password = updateAdmin.password ? updateAdmin.password : admin.password;
+      return res.status(200).json(admin);
```
Comment (NicoMunoz909) [inline controllers/admins.js:31]: All this block should have one more indentation

File: controllers/admins.js [hunk 7]
```diff
@@ -2,52 +2,41 @@ const fs = require('fs');
...
+    }
+  });
+  } 	return res.status(404).send("Admin not found");
```
Comment (NicoMunoz909) [inline controllers/admins.js:34]: This line would look better on the line below

File: controllers/admins.js [hunk 8]
```diff
@@ -2,52 +2,41 @@ const fs = require('fs');
...
+  };
```
Comment (NicoMunoz909) [inline controllers/admins.js:35]: This indentation should be removed

**Review by NicoMunoz909 — CHANGES_REQUESTED** [review]
Check indentation on lines 6, 12, 18

**Comment by nicolasblasco** [comment]
Problem with the last commit. You modified package-lock.json. We must no touch that file!!

File: package-lock.json [hunk 9]
```diff
@@ -1,1762 +1,8 @@
 {
   "name": "basd-mindset",
   "version": "0.0.1",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
   "requires": true,
```
Comment (NicoMunoz909) [inline package-lock.json:5]: this file shouldn't be modified (!)

**Review by NicoMunoz909 — CHANGES_REQUESTED** [review]

**Review by NicoMunoz909 — APPROVED** [review]

**Pull request merged**